jsonschema-core.xml

```diff
@@ -1163,37 +1167,19 @@
                         media type "application/schema+json".
                     </t>
                     <t>
-                        The "$schema" keyword SHOULD be used in a resource root schema.
-                        It MUST NOT appear in resource subschemas.  If absent from the root schema, the
-                        resulting behavior is implementation-defined.
+                        The "$schema" keyword SHOULD be used in the document root schema object,
+                        and MAY be used in the root schema objects of embedded schema resources.
+                        It MUST NOT appear in subschemas.  If absent from the document root schema,
```

Comment:
the wording of "subschemas" is a bit confusing to me. if I can attempt to clarify with an example

```yaml
$schema: draft7
$id: root
items:
  $schema: draft2020
  $id: items
```

items is a root schema object because it's got an $id. but is items no longer a "subschema" of the root? I feel like saying it's not a subschema isn't consistent with how the term subschema is used in the rest of the spec.

Reply:
Maybe "non-root schemas"?  Or just "other schemas"?

Reply:
(IMO) "subschema" only has context when in reference to another schema.  the schema at id "items" is a subschema of the schema at id "root". The schema at id "root" is not a subschema of anything (it has no parent schema).

Reply:
Historically, we've used "subschema" to indicate containment but not reference.  Referenced schemas have not been historically labelled "subschemas."

Reply:
Maybe it comes down to how we define a "schema document."  Is that the specific file, and that's it?  Or is it the file, and all of its external references?  (This pertains to the change below as well.)

Reply:
I would define a "document" as a self-contained JSON data structure (that might have originated from a network query, or from disk, or just being supplied to the evaluator directly). It might have $refs to external resources but those are separate documents.  i.e. this is the thingy that I would put into an $id registry that other schemas can '$ref'er to (but there could be other $ids in subschemas in this document that also get registered, so it's not a 1:1 relationship).

Reply:
@gregsdennis @karenetheridge "schema document" is defined in the spec.  It is defined to be the same as [JSON text](https://tools.ietf.org/html/rfc8259#section-2) from the JSON RFC (although I think it doesn't literally reference that).

Reply:
@gregsdennis @karenetheridge it might be useful to have a name for that "document plus all external references, transitively" concept.  I think I've talked about "schema sets" or "schema document sets" although probably not in the spec.  And not clearly distinguishing whether that's a complete reference graph, or a partial one, or multiple ones, or a random selection, etc.

Reply:
Agreed. Some implementations provide an interface to extract these - either as multiple documents or "bundled" together in one (potentially renaming conflicting $refs if needed). In one of my web apps I do this in a `GET /json_schema/:schema_name` endpoint.

Reply:
@karenetheridge yeah, the bundling use case was ultimately what we decided use when figuring out what to do with `$id` (splitting the `$anchor` case out and cutting a bunch of nonsensical but syntactically legal values).  And that led to the idea of `$id` as identifying _resources_ as opposed to just random otherwise unremarkable schema objects.

Reply:
> it might be useful to have a name for that "document plus all external references, transitively" concept
- @handrews 

I would call this a "trancluded de-referenced bundle".
- Transclusion is what is done to the schemas
- De-referenced is the result of the process
- Bundle is the end product descriptor

Reply:
@Relequestual I'll think on this.  Does it need to go in now or can we file an issue for this terminology?  If we adopt it (and I'm cautiously supportive), it should probably go in everywhere and I'd rather not add all of that in this PR.

Reply:
OK I've gone back over all of this.  I am about to push a commit to address @notEthan's original question about the usage of "subschema" (I agree it is unclear).

For the transcluded de-reference bundle thing, I have filed issue #935.  Note that the discussion involved a lot more than the bundle use case, so it really needs to be discussed separately from this PR.  We can add more terminology later, assuming my most recent commit addresses the actual confusion in the PR text.